tests/test_registry.py

```diff
@@ -380,13 +380,9 @@ def test___repr__(self):
             "<Introspectable category 'category', discriminator 'discrim'>",
         )
 
-    def test___nonzero__(self):
-        inst = self._makeOnePopulated()
-        self.assertEqual(inst.__nonzero__(), True)
-
     def test___bool__(self):
         inst = self._makeOnePopulated()
-        self.assertEqual(inst.__bool__(), True)
+        self.assertEqual(bool(inst), True)
```

Comment:
this is a dangerous change to the test because any object instance will test true if it doesn't override `__bool__`... I think it'd be better to keep it the way it was and explicitly check that it is defining the protocol